internal/developmentmembership/trackingsubjectset.go

```diff
@@ -100,9 +100,7 @@ func (tss *TrackingSubjectSet) getSetForKey(key string) datasets.BaseSubjectSet[
 			fs.excludedSubjects = excludedSubjects
 			fs.caveatExpression = caveatExpression
 			for _, source := range sources {
-				if source.relationships != nil {
```

Comment:
Why isn't this check necessary  anymore?

Reply:
I moved it into the set impl